### PR TITLE
Hc starting on UI

### PR DESCRIPTION
**Pull request by hcontr4**
Commits up to end of part 3.

### CODE REVIEW

**Review by kcalva — APPROVED** [review]
Everything looks good. One thing to just keep in mind when continuing the next part is that you still have the <% notice %> on the scaffold generated models, make sure to delete them. Other than you are set

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,11 +6,24 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: Great job adding your CDN to a partial, this helps clean up the application layout file :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,11 +6,24 @@
...
+
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
-    <%= yield %>
+
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:17]: same as above

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,11 +6,24 @@
...
+
+    <% if notice.present? %>
+      <%= render partial: "shared/flash", locals: { message: notice, css_class: "success" } %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:20]: same as above

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    </button>
+
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Dropdown
+          </a>
+          <ul class="dropdown-menu">
+            <li><a class="dropdown-item" href="/photos">Photos</a></li>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:18]: remember we can start using helper methods here like link_to

File: app/views/shared/_navbar.html.erb [hunk 6]
```diff
@@ -0,0 +1,46 @@
+<nav class="navbar navbar-expand-lg bg-light mb-4">
+  <div class="container-fluid">
+    <!-- <a class="navbar-brand" href="#">Navbar</a> -->
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:3]: Try to delete commented out code if it is no longer being used

File: lib/tasks/dev.rake [hunk 7]
```diff
@@ -11,6 +11,20 @@ task({ :sample_data => :environment }) do
     User.destroy_all
   end
 
+  usernames = Array.new { Faker::Name.first_name }
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:19]: great job on your sample data task!

**Review by JoseMDO — APPROVED** [review]
Great job Herman, everything looks good!

**Review by rayguna — COMMENTED** [review]
The tag `<p style="color: green"><%= notice %></p>` could be deleted within:
- app/views/likes/show.html.erb
- app/views/follow_requests/show.html.erb
- app/views/comments/show.html.erb

**Comment by MauritzKurt** [comment]
Looks great, excellent job paying attention to details!

**Pull request merged**